scipy/stats/_hypotests.py

```diff
@@ -1144,6 +1144,10 @@ def boschloo_exact(table, alternative="two-sided", n=32):
 
     fisher_stat = pvalues[table[0, 0], table[0, 1]]
     index_arr = pvalues <= fisher_stat
+    index_arr = np.logical_or(
+        index_arr,
+        np.isclose(pvalues - fisher_stat, 0)
+    )  # Because the hypergeometric has some slight numerical errors
```

Comment:
I thought I'd link [the conversation](https://github.com/scipy/scipy/pull/14153/files/4d875dd80645147b256d9998533a55963e6c82b0#diff-6ba7fcf5a871c6742dd082e5fc1801ef3ce2126456b052e9d0a126e70aab5f99) to the new relevant line.

I understand that you included this so that the tests would pass. In the interest of decoupling the clear improvement to the hypergeometric distribution from this debatable change to `boschloo_exact`, I would suggest instead xfailing the problematic test case and opening a separate issue/PR for this change. 

If you really want to include this change to `boschloo_exact` here, please address the comment:
> Can you determine a theoretical bound on how close is close enough (i.e. definitely smaller than a legitimate difference between pvalues)? I think you can - it's something like the `hypergeom.pmf` near `fisher_stat`.

This is important, because `np.isclose` with default tolerances will not do the right thing when the difference between legitimately distinct elements of the distribution are small. 

I'll support the `boschloo_exact` changes, whether they're left in this PR or moved to a separate PR, but I will let @WarrenWeckesser and @mckib2 review the rest of this.

Reply:
I removed Boschloo Updates and add an `xfail` for the test which is sensitive to epsilon machine. I am going to submit an other PR to correct this. 
If this is ok for you this PR is ready to be merged according to @pabristow and @mckib2